site/docs/v1/tech/admin-guide/upgrade.adoc

```diff
@@ -4,44 +4,300 @@ title: Upgrade FusionAuth
+
+Because individual nodes are taken offline during the rolling update, the capacity of your installation is reduced during the upgrade.
+
+You can choose to run in-place upgrades on each node or to replace each node with a new node running the latest version of FusionAuth. In-place upgrades are potentially faster, but if something goes wrong, you may need to roll back the upgrade. Replacing nodes may be slower, but if something goes wrong, you can simply revert to the previous version of FusionAuth by pointing the load balancer to the old nodes.
+
+A rolling upgrade procedure might look like this:
+
+- Stop the first node and remove it from the load balancer.
+- Upgrade FusionAuth on the first node. Once FusionAuth is updated, start it up so that it can perform the database migrations. FusionAuth will lock the database and own it while the database upgrade is completed. Note that there will be a schema mismatch between the first node and the other nodes, so the other nodes may give errors while the first node is being upgraded. The nature of these errors depends on the code paths being updated as well as how users interact with the system. They should be minimal.
+- Start the first node and add it back to the load balancer.
+- Stop the second and third nodes, remove them from the load balancer, and upgrade them. 
+- Start the second and third nodes and add them back to the load balancer.
+
+There is still some downtime with this procedure, but it is limited to the time it takes to run the database migration on the upgraded first node. There will be limited capacity during the upgrade, especially while the second and third nodes are upgrading, but the system will be available for use for most of the upgrade process.
+
+++++
+{% plantuml source: _diagrams/docs/admin-guide/rolling_upgrade.plantuml, alt: "Rolling upgrade." %}
+++++
+
+=== Blue-Green Deployment
+
+A blue-green deployment is a technique that reduces downtime and risk by running two identical environments in parallel. A load balancer is used to direct traffic to one of the environments. Using a blue-green deployment instead of performing a rolling upgrade isolates the upgrade process from the production environment, reducing the impact and risk to the current production environment.
+
+To perform a blue-green deployment with a three-node cluster, first create three new nodes with the new version of FusionAuth on each of them. Start up one of the new nodes. When the new node starts and connects to the database, it will lock the database and own it while it completes the database upgrade. The original nodes might give errors as they will not have access to the database. These original nodes can be removed. Now add the new nodes to the cluster. The system will be down for the duration of the database upgrade but once the database upgrade is complete, the system will be back online with the upgraded version of FusionAuth on the new nodes. 
```

Comment:
```suggestion
To perform a blue-green deployment with a three-node cluster, first create three new nodes with the new version of FusionAuth on each of them. Start up one of the new nodes. When the new node starts and connects to the database, it will lock the database and own it while it completes the database upgrade. The original nodes might give errors as they will not have access to the database. These original nodes can be removed from the load balancer. Add the new nodes to the cluster. 

With this approach, the system will be down for the duration of the database upgrade but once the database upgrade is complete, the system will be back online with the upgraded version of FusionAuth on the new nodes. 
```